File: src/main/java/io/bloco/faker/components/Color.java
package io.bloco.faker.components;

import io.bloco.faker.FakerComponent;
import io.bloco.faker.FakerData;
import io.bloco.faker.helpers.MathHelper;

public class Color extends FakerComponent {

    public static final int MAX_RGB = 256;
    public static final double MAX_HSL = 360d;

    private MathHelper mathHelper;

    public Color(FakerData data) {
        super(data);
        mathHelper = new MathHelper();
    }

    public String hexColor() {
        return String.format("#%06x", randomHelper.number((int) Math.pow(MAX_RGB, 3)));
    }

    public String colorName() {
        return fetch("color.name");
    }

    public int singleRgbColor() {
        return randomHelper.number(MAX_RGB);
    }

    public int[] rgbColor() {
        return new int[]{singleRgbColor(), singleRgbColor(), singleRgbColor()};
    }

    public double singleHslColor() {
        return mathHelper.round(randomHelper.range(0, MAX_HSL), 2);
    }

    public double alphaChannel() {
        return randomHelper.randDouble();
    }

    public double[] hslColor() {
        return new double[]{singleHslColor(), singleHslColor(), singleHslColor()};
    }

    public double[] hslaColor() {
        return new double[]{singleHslColor(), singleHslColor(), singleHslColor(), alphaChannel()};
    }
}

File: lib/src/main/kotlin/io/bloco/faker/components/Color.kt
package io.bloco.faker.components

import io.bloco.faker.FakerComponent
import io.bloco.faker.FakerData
import io.bloco.faker.helpers.RandomHelper

class Color(data: FakerData) : FakerComponent(data) {
    fun hexColor(): String {
        return "#%06x".format(RandomHelper.number(MAX_RGB * MAX_RGB * MAX_RGB))
    }

    fun colorName(): String {
        return fetch("color.name")
    }

    fun singleRgbColor(): Int {
        return RandomHelper.number(MAX_RGB)
    }

    fun rgbColor(): IntArray {
        return IntArray(3) { singleRgbColor() }
    }

    fun singleHslColor(): Double {
        return RandomHelper.range(0.0, MAX_HSL)
    }

    fun alphaChannel(): Double {
        return RandomHelper.randDouble()
    }

    fun hslColor(): DoubleArray {
        return DoubleArray(3) { singleHslColor() }
    }

    fun hslaColor(): DoubleArray {
        return hslColor() + alphaChannel()
    }

    companion object {
        const val MAX_RGB = 256
        const val MAX_HSL = 360.0
    }
}
